File: Array/twoSums.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    const hashTable = {}; // created a hashTable to store the elements of the array
    for (let i = 0; i < nums.length; i++) {
        hashTable[nums[i]] = i; // function to store the elements of the arry in the hashtable
    }

    // itarate over the array and check if compliment of each element is in the hashTable
    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];
        if (complement in hashTable && hashTable[complement] !== i) {
            return [i, hashTable[complement]];
        }
    }
    return [] // if no solution return empty array
};

const nums = [2, 7, 11, 15];
const target = 17;

const result = twoSum(nums, target);

if (result !== null) {
    console.log(`The numbers that add up to ${target} are ${nums[result[0]]} and ${nums[result[1]]}`)
} else {
    console.log('No such pair')
}

File: Array/twoSums.py
#Given an array of integers nums and an integer target
#return indices of the two numbers such that they add up to target.
# Each input should have one solution and elements shoud not be repeated twice.
nums_array = [2,3,4,5]
target = 8
pairs = False
for  i in range(len(nums_array)):
    for j in range(i + 1, len(nums_array)):
        if nums_array[i] + nums_array[j] == target:
            pairs = True
            print(f"numbers are {nums_array[i]} + {nums_array[j]} = {target}")
if not pairs:          
        print("Invalid set of numbers")
        


            
    
    